From 0aa90b0c8ad0da06bd496dc5c3162e770b26a899 Mon Sep 17 00:00:00 2001
From: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>
Date: Thu, 4 Jan 2024 03:07:54 -0500
Subject: [PATCH] [Maint] Update build_docs workflow to match napari/docs
 (#6547)

This PR updates the build_docs workflow to match the latest napari/docs
workflow
Specifically add concurrency (napari/docs PR:
https://github.com/napari/docs/pull/277 ) and to fix graphical issues
related to the xvfb action ( napari/docs PR:
https://github.com/napari/docs/pull/294 )
---
 .github/workflows/build_docs.yml | 20 +++++++++++++-------
 1 file changed, 13 insertions(+), 7 deletions(-)

diff --git a/.github/workflows/build_docs.yml b/.github/workflows/build_docs.yml
index bbe7f123..48456cc1 100644
--- a/.github/workflows/build_docs.yml
+++ b/.github/workflows/build_docs.yml
@@ -11,6 +11,10 @@ on:
       - 'v*'
   workflow_dispatch:
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   build-and-upload-artifact:
     name: Build & Upload Artifact
@@ -24,7 +28,7 @@ jobs:
 
       - uses: actions/setup-python@v4
         with:
-          python-version: 3.10
+          python-version: "3.10"
           cache-dependency-path: |
             setup.cfg
             docs/requirements.txt
@@ -37,16 +41,18 @@ jobs:
           python -m pip install -e ".[docs]" -c "resources/constraints/constraints_py3.10.txt"
 
       - name: Build Docs
-        uses: aganders3/headless-gui@v1
+        uses: aganders3/headless-gui@v2
         env:
           GOOGLE_CALENDAR_ID: ${{ secrets.GOOGLE_CALENDAR_ID }}
           GOOGLE_CALENDAR_API_KEY: ${{ secrets.GOOGLE_CALENDAR_API_KEY }}
+          PIP_CONSTRAINT: ${{ github.workspace }}/napari/resources/constraints/constraints_py3.10_docs.txt
         with:
-          run: make docs
-
-      # TODO: enable when the link check passes
-      # - name: LinkCheck
-      #  run: make linkcheck-files
+          run:  make -C docs docs
+          # skipping setup stops the action from running the default (tiling) window manager
+          # the window manager is not necessary for docs builds at this time and it was causing
+          # problems with screenshots (https://github.com/napari/docs/issues/285)
+          linux-setup: "echo 'skip setup'"
+          linux-teardown: "echo 'skip teardown'"
 
       - name: Upload artifact
         uses: actions/upload-artifact@v3
-- 
2.34.1
